wearipedia/devices/fitbit/fitbit_sense_gen.py

```diff
@@ -0,0 +1,368 @@
+    int_delta = (delta.days * 24 * 60 * 60) + delta.seconds
+    random_second = randrange(int_delta)
+
+    start_time = start + timedelta(seconds=random_second)
+    end_time = start_time + timedelta(seconds=duration)
+
+    sleep_dict = {
+        "dateOfSleep": date,
+        "duration": duration * 1000,
+        "efficiency": np.random.randint(90, 99),
+        "endTime": str(end_time.isoformat()),
+        "infoCode": 0,
+        "isMainSleep": True,
+        "logId": np.random.randint(0, 1000000000),
+        "logType": "auto_detected",
+        "minutesAfterWakeup": round(duration / 60 * percents[2] / 100),
+        "minutesAsleep": round(duration / 60 * percents[0] / 100),
+        "minutesAwake": round(duration / 60 * percents[1] / 100),
+        "minutesToFallAsleep": round(duration / 60 * percents[3] / 100),
+        "startTime": str(start_time.isoformat()),
+        "timeInBed": round(duration / 60),
+        "levels": dict(),
+        "type": "stages",
+    }
+    sleep_dict["levels"]["data"] = []
+    sleep_dict["levels"]["shortData"] = []
+    sleep_dict["levels"]["summary"] = {
+        "deep": {"count": 0, "minutes": 0},
+        "light": {"count": 0, "minutes": 0},
+        "rem": {"count": 0, "minutes": 0},
+        "wake": {"count": 2, "minutes": round(duration / 60 * percents[1] / 100)},
+    }
+    #  core sleep levels
+    sleep_dict["levels"]["data"].append(
+        {
+            "dateTime": str(start_time.isoformat()),
+            "level": "wake",
+            "seconds": round(duration / 60 * percents[3] / 100) * 60,
+        }
+    )
+    # split sleep times
+    def split_the_duration(duration):
+        saver_of_duration = duration
+        while duration > 1:
+            n = np.random.randint(1, round(saver_of_duration / 12))
+            if duration - n >= 0:
+                yield n
+            else:
+                yield duration
+            duration -= n
+
+    generator = split_the_duration(round(duration * percents[0] / 100))
+
+    arr_of_durations = list(generator)
+
+    start_phases = start_time + timedelta(
+        seconds=round(duration / 60 * percents[3] / 100) * 60
+    )
+
+    # add phases
+    for i, item in enumerate(arr_of_durations):
+        if i != 0:
+            start_phases += timedelta(seconds=arr_of_durations[i - 1])
+
+        type = choice(["deep", "rem", "light"])
+
+        sleep_dict["levels"]["summary"][type]["count"] += 1
+        sleep_dict["levels"]["summary"][type]["minutes"] += round(item / 60)
+
+        sleep_dict["levels"]["data"].append(
+            {"dateTime": str(start_phases.isoformat()), "level": type, "seconds": item}
+        )
+
+    start_phases += timedelta(seconds=arr_of_durations[-1])
+    sleep_dict["levels"]["data"].append(
+        {
+            "dateTime": str(start_phases.isoformat()),
+            "level": "wake",
+            "seconds": round(duration / 60 * percents[2] / 100) * 60,
+        }
+    )
+
+    # do the same for wakeups
```

Comment:
Same as the other file, I think you should be able to combine this and the previous loop

Reply:
Please address this